### DISCUSSION TITLE
Calculation of negative damping in pitch mode of OC4 semi-5MW

### DISCUSSION THREAD

**Discussion by yangyangyang98** [OpenFAST/openfast#2192] [Q&A] [answered]
Dear @jjonkman :
     I want to calculate the negative damping of the OC4 semi-5MW， I adopt the simplified single degree of freedom floating body pitch mode model like this paper: #Influence of Control on the Pitch Damping of a Floating Wind Turbine
[Influence of Control on the Pitch.pdf](https://github.com/OpenFAST/openfast/files/15256758/Influence.of.Control.on.the.Pitch.pdf)
And I found most of the parameters in the report of “Definition of the 
Semisubmersible Floating System for Phase II of OC4”, except for the damping associated with hydrodynamic radiation in pitch. But I found the B55=0 at the platform-pitch natural frequency, Wxn=0.037Hz in page 18 of this report. So, I would like to verify if the damping associated with hydrodynamic radiation in pitch is zero. 
     Additionally, I found that the units given for damping coefficient and stiffness coefficient in this report are Nms^2/rad^2 and Nm/rad, instead of kgm^2/s and kgm^2/s^2. So I cannot derive that the natural frequency of the pitch mode of the semi OC4-5MW wind turbine is 0.037Hz, May I know how to convert units, or could you please tell me the converted values?
![single DOF pitch model](https://github.com/OpenFAST/openfast/assets/169322023/8b2c10d8-32ea-4e3b-8f3c-1e90549436c9)
![B55](https://github.com/OpenFAST/openfast/assets/169322023/58b5e7d7-eb26-49dd-b91b-aa208343d00a)

**Comment by jjonkman** [comment]
Dear @yangyangyang98,

I agree that the hydrodynamic radiation damping is quite small at 0.037 Hz based on Figure 4-3 of the OC4-DeepCwind semisubmersible specifications report.

I'm not sure I understand your question about units, but N*m/rad is equivalent to kg*m^2/s^2 and N*m/(rad/s) is equivalent to kg*m^2/s because 1 N = 1 kg*m/s^2 and rad is unitless.

Best regards,

**Reply by yangyangyang98** [reply]
Thank you for your reply！

But I still have some questions that I would like to ask you. I use the following formula to calculate the natural frequency and damping ratio of the floating pitch mode:
![negative damping](https://github.com/OpenFAST/openfast/assets/169322023/6e4821a5-ad27-4081-a35e-6dd24eed0518)

And I found most of the parameters in the report of “Definition of the Semisubmersible Floating System for Phase II of OC4”, I fonud I_Mass = 6.827*10^9 kgm^2; A_Radiation = 7.21*10^9 kgm^2; C_Hydrostatic = -3.776*10^8 Nm/rad; C_Line = 8.73*10^7 Nm/rad; As mentioned above, B_Radiation is almost equal to 0 at 0.037 Hz; I found the quadratic viscous drag coefficient of pitch = 3.7*10^10 Nms^2/rad^2 in Page24, but I couldn't find the linearized damping associated with hydrodynamic viscous drag in pitch B_Viscous。

Based on the above data, I calculated the stiffness term Kx to be negative. So how can I calculate to obtain the conclusion that the pitch mode frequency is 0.037Hz? In addition, what is the value of B_Viscous?

Looking forward to your reply.

**Reply by jjonkman** [reply]
Dear @yangyangyang98,

Just a few comments:

- The value you are using for I_Mass is not correct; I see that you are using the pitch inertia about the CM of the platform, but I_Mass should be the full-system pitch inertia about the platform reference point.
- The value you are using for C_Hydrostatic is missing the contribution of pitch restoring associated with full-system weight and its center of mass.
- B_Quad represents the quadratic drag in pitch.  You'd have to convert this to an equivalent linear drag, e.g., using Borgman linearization to calculate B_Viscous.

Best regards,

**Reply by yangyangyang98** [reply]
Dear @jjonkman :

I am a control engineer of wind turbine who is not familiar with the theory of hydrodynamics, which is far beyond my professional ability. I have been searching for the above parameters for a long time and have struggled to achieve the expected results. But no results. 

May I ask if it is convenient for you to directly provide the specific values of the above three parameters I_Mass、C_Hydrostatic and B_Quad of OC4-DeepCwind semisubmersible 5MW wind turbine? I would greatly appreciate it. Or if there are any specific parameters where I can reference them. 

Thank you again!

**Reply by jjonkman** [reply]
Dear @yangyangyang98,

Regarding I_Mass for the NREL 5-MW baseline wind turbine atop the OC4-DeepCwind semisubmersible, this is available in my post dated Feb 13, 2020 in the following topic on our forum: https://forums.nrel.gov/t/oc4-deepcwind-semisubmersible-wamit-files/2203.

Regarding C_Hydrostatic, this can be computed as -m * g * z_cg, where m is the full-system mass, g is gravity, and z_cg is vertical center of mass.  m * z_cg is also available in the aforementioned forum topic.

I have not personally computed B_Quad for the OC4-DeepCwind semisubmersible.

Best regards,

**Reply by yangyangyang98** [reply]
Dear @jjonkman :
Thank you for your reply.

I have carefully reviewed the content of the link you mentioned above. And it can seen that I_Mass=1.269 * 10 ^ 10kgm ². My understanding of the hydrodynamic stiffness coefficient KT is that the influence of the full system body weight / center of mass is should add to pitch mode stiffness, KT = C_Hydrostatic + C_Line - mgz = -3.77610^8 + 8.7310^7 -mgz，where m is the full-system mass, g is gravity, and z is the full-system center of mass location (negative valued)，The m = 1.37*10^7 kg, g=9.8m/s^2, but I cannot find the full-system center of mass location. But based on the pitch mode natural frequency of 0.037Hz, I can deduce that the center of mass position of the full system is -2.365m. May I ask if this value is correct？

Looking forward to your reply.

**Reply by jjonkman** [reply, marked as answer]
Dear @yangyangyang98,

I agree with your determination of I_Mass.

The 6x6 mass matrix contains the mass (upper-left quadrant), inertias (lower-right quadrant) and mass times center of mass (off-diagonal quadrants).  The mass (m) reported in that forum post is 1.407E+07 kg and the vertical center of mass (z_cg) is -1.392E+08 / 1.407E+07 = -9.893 m.

Best regards,

**Reply by yangyangyang98** [reply]
Dear @jjonkman :

Thank you for your early reply! It is a good time to communicate with you!

Based on the data you provided above, Based on the data you provided, I can calculate the natural frequency of the pitch mode of the OC4 semi submersible FOWT as W_xn = sqrt( K_x /M_x ) = sqrt(  (C_Hydrostatic+C_Line-mgz)/ (I_Mass+A_Radiation) ) =  sqrt(  (-3.776*10^8+8.73*10^7-1.407*7*9.8*(-9.893))/ (1.269*10^10+7.21*10^9) ) = 0.23229 rad/s, this is consistent with the 0.037Hz in the report of “Definition of the Semisubmersible Floating System for Phase II of OC4”. That's really good news！

Thank you again！

Besides, I still have doubts about the calculation of FOWT pitch mode damping, mainly because the value of B_Viscous cannot be obtained. I found the quadratic viscous drag coefficient of pitch = 3.7*10^10 Nms^2/rad^2 in Page24 of “Definition of the Semisubmersible Floating System for Phase II of OC4”, but I couldn't find the linearized damping associated with hydrodynamic viscous drag in pitch B_Viscous. Do you have any suggestions for this？

**Reply by jjonkman** [reply]
Dear @yangyangyang98,

As I mentioned before, you could use e.g., Borgman linearization to compute B_Viscous, but I have not done this myself to provide detailed guidance.

Best regards,

**Reply by yangyangyang98** [reply]
Dear @jjonkman :

Thank you for your reply. I will continue to study according to your suggestions.

**Comment by yangyangyang98** [comment]
Dear @jjonkman :
I have successfully calculated the pitch mode damping of the NREL 5MW semi submersible wind turbine with your help . Thank you very much! But now I need to further calculate the pitch mode damping of the IEA 15MW semi submersible wind turbine, but I have encountered some parameter issues and would like to seek your help. Below, I will describe my problem in detail,

 I still adopt the simplified single degree of freedom floating body pitch mode model like this paper:#Influence of Control on the Pitch Damping of a Floating Wind Turbine
[Influence of Control on the Pitch.pdf](https://github.com/OpenFAST/openfast/files/15256758/Influence.of.Control.on.the.Pitch.pdf)
![single DOF pitch model](https://github.com/user-attachments/assets/9942635c-55c0-47b6-81db-2811172212cb)

 I have found these the parameters: ARadiation(added inertia) is 1.25*1010kg•m2; BRadiation(the damping associated with hydrodynamic radiation in pitch) is near 0 at the platform-pitch natural frequency Wxn=0.213rad/s; BViscous(the linearized damping associated with hydrodynamic viscous drag in pitch) is 1.676*1010 Nms2/rad2；total system mass is 20,093t in the report of "Definition of the UMaine VolturnUS-S Reference Platform Developed for the IEA Wind 15-Megawatt Offshore Reference Wind Turbine".

I would like to ask if the above parameters are correct？Besides, I would like to consult with you that the values of these parameters: the IMass(pitch inertia associated with wind turbine and floating)、CHydrostatic(hydrostatic restoring in pitch)、CLines(the linearized hydrostatic restoring in pitch from all mooring lines) and z_cg(vertical center of mass) .

Looking forward to your early reply.

**Reply by jjonkman** [reply]
Dear @yangyangyang98,

I was not the one who made the OpenFAST model of the IEA Wind 15-MW RWT atop the VolturnUS semisubmersible and it would take some work to derive these parameters from the documented information.  Perhaps someone more experienced with this model can reply, but it is probably better to post your question in the IEA Wind 15-MW RWT repository: https://github.com/IEAWindTask37/IEA-15-240-RWT/issues.

Best regards,

**Reply by yangyangyang98** [reply]
Dear @jjonkman :
Thank you for your early reply. 

I carefully reviewed the previous reply. And found that some parameters can be obtained from the FRC file, which is located at in the WAMIT file of the Hydrodata module for IEA-15-240-RWT(https://github.com/IEAWindTask37/IEA-15-240-RWT/blob/master/OpenFAST/IEA-15-240-RWT-UMaineSemi/HydroData/wamit_inputs_2ndOrder/IEA-15-240-RWT_2nd.frc) .The following two images are NREL 5MW and IEA 15MW respectively FRC file.
![5MW](https://github.com/user-attachments/assets/61b282e1-f57f-4b2d-83df-14acc521933e)
![15MW](https://github.com/user-attachments/assets/56db44fe-9fbc-4bf0-aa4e-eab4bb03cd9e)


Based on your previous response and the comparative analysis of these two pictures, I can infer that IMass(pitch inertia associated with wind turbine and floating) is 4.26*10e10kg•m2、CLines(the linearized hydrostatic restoring in pitch from all mooring lines) is 1.69*10e8Nm/rad; total system mass is 2.01e7kg; and z_cg(vertical center of mass) is -4.66e7/2.01e7=-2.322m.  The CHydrostatic(hydrostatic restoring in pitch) is 2.190*109Nm/rad, which can be found in "Definition of the UMaine VolturnUS-S Reference Platform Developed for the IEA Wind 15-Megawatt Offshore Reference Wind Turbine". 

Now my question is
1. What does the first element in the damping matrix of aboved FRC file(the parameters marked with arrows) ？Is it BViscous(the linearized damping associated with hydrodynamic viscous drag in pitch)？
2. The quadratic hydrodynamic viscous drag in pitch is 1.676E+10N-m-s2, which can be found in "Definition of the UMaine VolturnUS-S Reference Platform Developed for the IEA Wind 15-Megawatt Offshore Reference Wind Turbine"，But it doesn't seem to be equal to BViscous?

Looking forward to your early reply.

**Reply by jjonkman** [reply]
Dear @yangyangyang98,

Again, I can't vouch for the correctness of all these values for the IEA Wind 15-MW RWT atop the VoltunUS semisubmersible, but I agree your interpretation of pitch inertia, total mass, and center of mass based on IMASS and C_Lines based on ISTIF.  Please note that value of C_Hydrostatic does not include the contribution of body weight (based on the full-system mass and center of mass) as stated in the definition report you referred to, so, that pitch restoring should be added separately.

Regarding your questions 1-2, you are pointing to an element of ISTIF, not IDAMP, which has been set to zero.  I agree that B_Viscous is not the same as the specified viscous drag, which is expressed quadratically.  B_Viscous expresses a linearized form of this viscous drag.

Best regards,

**Reply by yangyangyang98** [reply]
Dear @jjonkman :
Thank you for your early reply.

I haved noted that value of C_Hydrostatic does not include the contribution of body weight (based on the full-system mass and center of mass) in the definition report "Definition of the Semisubmersible Floating System for Phase II of OC4", as shown in the following figure：
![CHydro_5MW](https://github.com/user-attachments/assets/a6282f83-9594-42b5-aad0-4b4edda4e8fa)
However, the C_Hydrostatic in the IEA 15MW report “Definition of the UMaine VolturnUS-S Reference Platform Developed for the IEA Wind 15- Megawatt Offshore Reference Wind Turbine ” seems to already include the contribution of body weight (based on the full-system mass and center of mass)，which is calculated by using WAMIT. As shown in the following figure:
![CHydro_15MW](https://github.com/user-attachments/assets/5eb56bce-eb22-48e1-8b8c-ce256024ae7f)
I don't know if my understanding about IEA 15MW report is correct.

Looking forward to your early reply.

**Reply by jjonkman** [reply]
Dear @yangyangyang98,

As the IEA Wind 15-MW specifications report says, "The mass component of each hydrostatic restoring moment...has not been included here", which is referring to Table 3.

Best regards,

**Reply by yangyangyang98** [reply]
Dear @jjonkman :

Thank you for your early reply. Your reply is helpful. I will conduct further research.

Best regards
